File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        l1=len(word1)
        l2=len(word2)
        result=[]

        for w1, w2 in zip(word1, word2):
            result.append(w1)
            result.append(w2)

        
        return "".join(result + [word1[len(word2):]] + [word2[len(word1):]])
            

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string merge;
        merge.reserve(word1.length() + word2.length());

        for(int i=0; i<max(word1.length(),word2.length());i++){
            if(i<word1.length())
            {
                merge += word1[i];
            }
            if(i<word2.length()){
                merge += word2[i];
            }
            
        }
        return merge;
    }
};